### ISSUE TITLE
After build smoke tests

### ISSUE THREAD

**Issue by grantmacken** [zadean/xqerl#15]
I've open this issue to discuss *after build* smoke tests.
This could be done with travis.yml file  
 - build docker image from repo after each branch commit/push
 - run newly built container
 - do smoke tests in the form of user orientated tasks in script section 

As rough outline
the smoke test should cover what a user should be able to do
- [x] run a simple xquery expression 'xqerl:run()'
- [x ] compile a xquery 'main' file and run the query xqerl:compile()
- [x ] compile a xQuery library file 
- [ ] insert and delete XML docs into and from database.
- [ ] insert binary, text and json files into database
- [ ] via curl, use each the rest methods to GET, POST, PUT, DELETE database resources via restXQ endpoint

and anything else the user should be able to do.

**Comment by zadean** [comment]
Good idea and something I've been hoping to get to but haven't. I think all the DB actions could be  done using the common_test framework.  This would just mean making the test suites.

Another good smoke test would be a dialyzer run to look for broken or dead code.
